src/sentry/api/endpoints/user_notification_options.py

```diff
@@ -0,0 +1,61 @@
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.api.serializers import serialize
+from sentry.api.validators.notifications import validate_type
+from sentry.models import User
+from sentry.models.notificationsettingoption import NotificationSettingOption
+from sentry.notifications.serializers import NotificationSettingsOptionSerializer
+from sentry.notifications.validators import UserNotificationSettingOptionWithValueSerializer
+
+
+@control_silo_endpoint
+class UserNotificationOptionsEndpoint(UserEndpoint):
+    # TODO(Steve): Make not private when we launch new system
+    private = True
+
+    def get(self, request: Request, user: User) -> Response:
+        """
+        Retrieve the notification preferences for a user.
+        Returns a list of NotificationSettingOption rows.
+        """
+        notification_type = request.GET.get("type")
+        notifications_settings = NotificationSettingOption.objects.filter(user_id=user.id)
+        if notification_type:
+            try:
+                validate_type(notification_type)
+            except ParameterValidationError:
+                return self.respond({"type": ["Invalid type"]}, status=status.HTTP_400_BAD_REQUEST)
+            notifications_settings = notifications_settings.filter(type=notification_type)
+
+        notification_preferences = serialize(
+            list(notifications_settings), request.user, NotificationSettingsOptionSerializer()
```

Comment:
i don't believe you have to provide a serializer here as you've registered the object

Reply:
@JoshFerge there are some weird circular dependencies when I do this, I'm just going to not register it and use it directly